File: decisionTree.py
# -*- coding: utf-8 -*-
"""
Created on Sat Jan  4 23:51:34 2020

@author: 138709
"""

# Decision Tree Classification

# Import the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


# Import the dataset
dataset = pd.read_csv('Social_Network_Ads.csv')
X = dataset.iloc[:,[2,3]].values
Y = dataset.iloc[:,4].values


# Split the dataset into train and test sets
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.25, random_state = 0)


# Fitting the Decision Tree Model here
from sklearn.tree import DecisionTreeClassifier
model = DecisionTreeClassifier(criterion = 'entropy', random_state = 0)
model.fit(X_train, Y_train)


# Predicting the test set results
Y_pred = model.predict(X_test)


# Making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(Y_test, Y_pred)

File: decisionTree.R
# Decision Tree Classification
install.packages('caTools')
library(caTools)
install.packages('rpart')
library(rpart)

# Import the dataset
dataset = read.csv('Social_Network_Ads.csv')
dataset = dataset[3:5]


# Split the dataset into train and test sets
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.75)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


# Fit the Decision Tree model here
model = rpart(formula = Purchased ~ ., data = training_set, method = 'class')


# Predicting the test set results
Y_pred = predict(model, newdata=test_set[-3] , type = 'class')


# Making the confusion matrix
cm = table(test_set[,3], Y_pred)



# Plotting the Decision Tree
plot(model)
text(model)



